python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+            continue
+
+        file = {
+            "schema_year": schema_year,
+            "table_name": table_name,
+            "path": os.path.join(FILE_DIR, filename),
+            "extract_id": extract_id,
+        }
+        files_todo.append(file)
+    return files_todo
+
+
+def load_csv(filename):
+    with open(filename, "r") as fin:
+        reader = csv.DictReader(fin)
+        return [row for row in reader]
+
+
+def lower_case_keys(list_of_dicts):
+    return [{key.lower(): value for key, value in row.items()} for row in list_of_dicts]
+
+
+def remove_unsupported_columns(rows, table_name):
+    col_names_to_keep = [
+        col["name"]
+        for col in COLUMNS
+        if col["table_name"] == table_name and col["source"] == "cris"
+    ]
+    return [
+        {key: val for key, val in row.items() if key in col_names_to_keep}
+        for row in rows
+    ]
+
+
+def handle_empty_strings(rows):
+    for row in rows:
+        for key, val in row.items():
+            if val == "":
+                row[key] = None
+    return rows
+
+
+def handle_crash_date_time(rows):
+    """Combine crash_date and crash_time and format as ISO string.
+
+    The same ISO timestamp is stored as `crash_date` and `crash_time`
+    """
+    tzinfo = ZoneInfo("America/Chicago")
+    for row in rows:
+        # parse a string that looks like '12/22/2023'
+        crash_date = row["crash_date"]
+        month, day, year = crash_date.split("/")
+        # parse a string that looks like '12:19 PM'
+        crash_time = row["crash_time"]
+        hour_minute, am_pm = crash_time.split(" ")
+        hour, minute = hour_minute.split(":")
+        if am_pm.lower() == "pm":
+            hour = int(hour) + 12 if hour != "12" else int(hour)
+        else:
+            hour = int(hour) if hour != "12" else 0
+        # create a tz-aware instance of this datetime
+        crash_date_dt =  datetime(int(year), int(month), int(day), hour=hour, minute=int(minute), tzinfo=tzinfo)
+        # save the ISO string with tz offset
+        crash_date_iso = crash_date_dt.isoformat()
+        row["crash_date"] = crash_date_iso
+        row["crash_time"] = crash_date_iso
+
+
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def main():
+    overall_start_tme = time.time()
+    files_todo = get_files_todo()
+    extract_ids = list(set([f["extract_id"] for f in files_todo]))
+    # assumes extract ids are sortable oldest > newest by filename. todo: is that right?
+    extract_ids.sort()
+    for extract_id in extract_ids:
+        # get schema years that match this extract ID
+        schema_years = list(
+            set([f["schema_year"] for f in files_todo if f["extract_id"] == extract_id])
+        )
+        schema_years.sort()
+        for schema_year in schema_years:
+            for table_name in ["crashes", "units", "persons"]:
```

Comment:
will be rly important that this order never changes or triggers could break, maybe that would be a good comment to have as a reminder?